test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1415 @@
+}
+
+function checkEquality(
+  actualOnChainPayment: any,
+  expectedPayment: TestPayment,
+  paymentIndex: number
+) {
+  if (expectedPayment.status == PaymentStatus.Nonexistent) {
+    expect(actualOnChainPayment.account).to.equal(
+      ethers.constants.AddressZero,
+      `payment[${paymentIndex}].account is incorrect`
+    );
+    expect(actualOnChainPayment.amount).to.equal(
+      0,
+      `payment[${paymentIndex}].amount is incorrect`
+    );
+    expect(actualOnChainPayment.amount).to.equal(
+      0,
+      `payment[${paymentIndex}].status is incorrect`
+    );
+    expect(actualOnChainPayment.revocationCounter).to.equal(0);
+  } else {
+    expect(actualOnChainPayment.account).to.equal(
+      expectedPayment.account.address,
+      `payment[${paymentIndex}].account is incorrect`
+    );
+    expect(actualOnChainPayment.amount).to.equal(
+      expectedPayment.amount,
+      `payment[${paymentIndex}].amount is incorrect`
+    );
+    expect(actualOnChainPayment.status).to.equal(
+      expectedPayment.status,
+      `payment[${paymentIndex}].status is incorrect`
+    );
+    expect(actualOnChainPayment.revocationCounter).to.equal(
+      expectedPayment.revocationCounter || 0,
+      `payment[${paymentIndex}].revocationCounter is incorrect`
+    );
+  }
+}
+
+describe("Contract 'CardPaymentProcessorUpgradeable'", async () => {
+  const REVERT_MESSAGE_IF_CONTRACT_IS_ALREADY_INITIALIZED = 'Initializable: contract is already initialized';
+  const REVERT_MESSAGE_IF_CALLER_IS_NOT_OWNER = "Ownable: caller is not the owner";
+  const REVERT_MESSAGE_IF_CALLER_IS_NEW_REVOCATION_COUNTER_MAXIMUM_VALUE_IS_ZERO =
+    "CardPaymentProcessor: new value of the revocation counter maximum must be greater than 0";
+  const REVERT_MESSAGE_IF_CONTRACT_IS_PAUSED = 'Pausable: paused';
+  const REVERT_MESSAGE_IF_PAYMENT_AMOUNT_IS_ZERO = 'CardPaymentProcessor: payment amount must be greater than 0';
+  const REVERT_MESSAGE_IF_INPUT_ARRAY_OF_AUTHORIZATION_IDS_IS_EMPTY =
+    'CardPaymentProcessor: input array of authorization IDs is empty';
+  const REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO =
+    'CardPaymentProcessor: authorization ID must not equal 0';
+  const REVERT_MESSAGE_IF_TOKEN_TRANSFER_AMOUNT_EXCEEDS_BALANCE = "ERC20: transfer amount exceeds balance";
+  const REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_ALREADY_EXISTS =
+    'CardPaymentProcessor: payment with the provided authorization ID already exists and was not revoked';
+  const REVERT_MESSAGE_IF_ACCOUNT_IS_NOT_WHITELISTED = 'Whitelistable: account is not whitelisted';
+  const REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST =
+    'CardPaymentProcessor: payment with the provided authorization ID does not exist';
+  const REVERT_MESSAGE_IF_PAYMENT_IS_CLEARED =
+    'CardPaymentProcessor: payment with the provided authorization ID is cleared'
+  const REVERT_MESSAGE_IF_PAYMENT_IS_UNCLEARED =
+    'CardPaymentProcessor: payment with the provided authorization ID is uncleared'
```

Comment:
Codacy has a fix for the issue: [Replace `'CardPaymentProcessor:·payment·with·the·provided·authorization·ID·is·uncleared'` with `"CardPaymentProcessor:·payment·with·the·provided·authorization·ID·is·uncleared";`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
    "CardPaymentProcessor: payment with the provided authorization ID is uncleared";
```